#194 Basic solution to fix replay buffer settings set to 1

ThimoDEV opened the pull request:
Addresses #194 

Sets a default execution order attribute for all listeners and variables so that Listeners OnEnable() runs before Variables OnEnable().

This should allow new created event replay buffer settings be initialized to 0.

AdamRamberg commented:
Thanks @ThimoDEV for submitting this PR! 🙏

We need to do 2 things before approving. First, we need to address Variable Instancers in this PR and set their DefaultExecutionOrder. Secondly, we should also add a preference field (AtomsPreferences.cs) which specifies default buffer size on events.

ThimoDEV commented:
@AdamRamberg 

Changed the default execution order code because Runtime.cs had already constants for the variable instancers and I added the variable and listener ones there. Added a new Atoms setting to set a global replay buffer value. For now it doesnt check the set value. Any best suggestion on how to do this. I've been thinking about the following:

Add a regex check to restrict user input.

ThimoDEV commented:
@AdamRamberg 

Made changes based on your suggestions, there is now a global slider for the replay buffer. The only thing that is missing is a label at the end of the slider to show the current value. You know how I can add that?

AdamRamberg commented on Packages/Core/Runtime/Events/AtomEvent.cs:
```diff
@@ -31,10 +31,15 @@ public class AtomEvent<T> : AtomEventBase
 
+        private void OnEnable()
+        {
+            _replayBufferSize = AtomPreferences.ReplayBufferSize;
```
This will just overwrite the replay buffer size if set from the Inspector. Not 100% sure how to do this properly though.

AdamRamberg reviewed (changes requested):
This PR breaks the InfiniteWaves demo, since Listeners are running their OnEnable before Variable / Event Instancers are running. This causes errors when Listeners are trying to retrieve Events from the Instancers. Listeners should be executed after all others, not sure though if that might have other consequences. 

For this to be merged I need 2 changes: 
- Fix the assignment of the default preference value.
- InfiniteWaves demo can't break because of this change. 

About your question regarding the slider. Not sure about getting the value as a label next to the slider. I think however that the slider is good enough as it is.

miikalo commented:
The InfiniteWaves demo produces no errors, but it also does not spawn any enemies.

ThimoDEV commented:
My apologies, I changed the events by setting the use local boolean and set the replay buffer to their original value. I see now these event files conflict, but I cant see what their conflict is, or how should I do that?

Casey-Hofland commented:
Since this branch is broken and outdated, I will be closing it, as was discussed in #263 and #264.
